src/main/java/io/confluent/idesidecar/restapi/credentials/OAuthCredentials.java

```diff
@@ -55,11 +57,19 @@ public record OAuthCredentials(
   @Null
   Integer connectTimeoutMillis,
 
+  @Schema(description = "Additional property that can be added in the request header to identify "
+      + "the logical cluster ID to connect to. For example, this may be a Confluent Cloud " +
+      "Kafka or Schema Registry cluster ID.")
+  @JsonProperty(value = "logical_cluster_id")
+  @Null
+  String logicalClusterId,
```

Comment:
Do you know if the `logicalClusterId` is needed for clusters other than CCloud ones? If not, what about modeling this in a bit more generic way so that we can maybe support other `extension_`s as well without having to introduce new fields to the `OAuthCredentials`?

Reply:
Hm, yeah it is definitely CCloud specific, but so is the other `identityPoolId` field. 

> If not, what about modeling this in a bit more generic way so that we can maybe support other extension_s as well without having to introduce new fields to the OAuthCredentials?

Hm, would it be fine to revisit this at a later point? For now I imagine that we will have some client-side logic that will only surface these fields for direct connections to Confluent Cloud.

Reply:
Thanks for the clarification! From my POV, it's fine to add these two fields now because we probably won't need to support any other _custom_ JAAS configs in the near/mid term.

> Hm, would it be fine to revisit this at a later point?

I don't think we can remove them at a later point as they'll be part of the API contract and Direct Connection form. Removing them would have an impact on existing Direction Connections of users and be a breaking change.

> For now I imagine that we will have some client-side logic that will only surface these fields for direct connections to Confluent Cloud.

Maybe make it clear that these fields are only relevant for Direct Connections that target CCloud clusters by, for instance, adding the prefix `ccloud_` to their names?

Reply:
> I don't think we can remove them at a later point as they'll be part of the API contract and Direct Connection form. Removing them would have an impact on existing Direction Connections of users and be a breaking change.

Makes sense. Thanks for mentioning.

> adding the prefix ccloud_ to their names?

Prefixing `ccloud_` sounds good. :+1: